src/provider/utils.ts

```diff
@@ -31,6 +31,27 @@ export function txIdentifier(txHash?: BigNumberish, txId?: BigNumberish): string
 // hex string and BN are detected as block hashes
 // decimal string and number are detected as block numbers
 // null appends nothing to the request url
```

Comment:
these comments belong to the `BlockIdentifier` below... please rearrange this
@tabaktoni can add more comments about this approach. I generally would avoid suing new() inside of methods and use dependency injection... if it is even possible here